lib/p2p/NodeList.ts

```diff
@@ -170,23 +170,26 @@ class NodeList extends EventEmitter {
     const node = this.nodes.get(nodePubKey);
```

Comment:
I noticed `class NodeList` doesn't have any tests. Perhaps this is a good spot to add an integration test to `public addReputationEvents()`?

Reply:
I'll handle this in a separate PR.